src/app/dataset-view/additional-components/metadata-component/components/add-polling-source/steps/base-step/base-step.component.ts

```diff
@@ -66,7 +96,8 @@ export class BaseStepComponent extends BaseComponent implements OnInit {
                             ].includes(key),
                     )
                     .forEach((item: string) => {
-                        this.sectionForm.removeControl(item);
+                        if (item !== "eventTime")
```

Comment:
Please try to move all field-specific cleanup to the service. At least, declare constant for "special" field.